### ISSUE TITLE
Can update gradle wrapper with "any" version

### ISSUE THREAD

**Issue by StefMa** [gradle/gradle#6137]
Currently it is possible to run the `wrapper` task with **any version** I like.
E.g. `gradle wrapper --gradle-version 6.6.6`.
The task is *successfully* and replace all the files:
```
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

        modified:   gradle/wrapper/gradle-wrapper.jar
        modified:   gradle/wrapper/gradle-wrapper.properties
        modified:   gradlew
        modified:   gradlew.bat

no changes added to commit (use "git add" and/or "git commit -a")
```
When I now run gradle it fails. Because the verision does not exist.

### Expected Behavior
The wrapper task should check if the given version could be possibly downloaded.
If yes -> replace all the wrapper files
If not -> fail

### Current Behavior
The wrapper task succeeded without errors.
All the wrapper files will be replaced...

### Context
I accidentally updated my wrapper with `--gradle-version 4.9.1` (instead of 4.8.1).
That was a mistake which I haven't seen. The next gradle run leads to an crazy error:
<img width="1135" alt="bildschirmfoto 2018-07-31 um 11 53 51" src="https://user-images.githubusercontent.com/10229883/43452724-6d6475cc-94b8-11e8-830f-98f5ac1c2028.png">

### Steps to Reproduce (for bugs)
Run `gradle wrapper --gradle-version 6.6.6` in any gradle project.

### Your Environment
I think a build scan is not required.
macOS: 10.13.6
(More information needed?)

**Comment by StefMa** [comment]
I think this still valid.
Will check this tomorrow again.

Nevertheless. There are a few comments about a possible solution in this PR:
https://github.com/gradle/gradle/pull/8406
If someone wants to contribute in this, feel free to checkout the comments
from the PR.

**Comment by ov7a** [comment]
Fixed in 8.2: https://docs.gradle.org/8.2/release-notes.html#wrapper-task-validates-distribution-url
